### PR TITLE
Lk/port1.3 fix new

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
Adding LaplacianPyramid and LaplacianReconstruct

### CODE REVIEW

File: amd_openvx/openvx/CMakeLists.txt [hunk 1]
```diff
@@ -68,7 +68,7 @@ add_library(openvx SHARED ${SOURCES})
 add_library(vxu SHARED api/vxu.cpp)
 
 if (OpenCL_FOUND)
-	target_compile_definitions(openvx PUBLIC ENABLE_OPENCL=1)
```
Comment (kiritigowda) [inline amd_openvx/openvx/CMakeLists.txt]: turn on by default

File: amd_openvx/openvx/ago/ago_interface.cpp [hunk 2]
```diff
@@ -206,9 +206,8 @@ int agoOptimizeGraph(AgoGraph * agraph)
 		CAgoLock lock2(agraph->ref.context->cs);
 
 		// run DRAMA graph optimizer
-		agraph->status = agoOptimizeDrama(agraph);
+		agraph->status = agoOptimizeDrama(agraph);\
```
Comment (kiritigowda) [inline amd_openvx/openvx/ago/ago_interface.cpp]: check this

File: amd_openvx/openvx/ago/ago_kernel_api.cpp [hunk 3]
```diff
@@ -2215,9 +2215,8 @@ int ovxKernel_NonLinearFilter(AgoNode * node, AgoKernelCommand cmd)
 			return VX_ERROR_INVALID_FORMAT;
 		else if (!width || !height)
 			return VX_ERROR_INVALID_DIMENSION;
-		else if (node->paramList[3]->u.img.format != format) {
+		else if (node->paramList[3]->u.img.format != format)
```
Comment (kiritigowda) [inline amd_openvx/openvx/ago/ago_kernel_api.cpp:2218]: keep the brackets constant across

File: amd_openvx/openvx/ago/ago_kernel_api.cpp [hunk 4]
```diff
@@ -19703,8 +19756,10 @@ int agoKernel_LaplacianPyramid_DATA_DATA_DATA(AgoNode * node, AgoKernelCommand c
 	}
 #endif
 	else if (cmd == ago_kernel_cmd_query_target_support) {
-		node->target_support_flags = 0;
-		status = VX_SUCCESS;
+		node->target_support_flags = 0
+                    | AGO_KERNEL_FLAG_DEVICE_CPU              
+                    ;
```
Comment (kiritigowda) [inline amd_openvx/openvx/ago/ago_kernel_api.cpp:19770]: check this

File: amd_openvx/openvx/api/vx_api.cpp [hunk 5]
```diff
@@ -6733,10 +6734,11 @@ VX_API_ENTRY vx_status VX_API_CALL vxQueryPyramid(vx_pyramid pyr, vx_enum attrib
 VX_API_ENTRY vx_image VX_API_CALL vxGetPyramidLevel(vx_pyramid pyr, vx_uint32 index)
 {
 	AgoData * data = (AgoData *)pyr;
+
 	AgoData * img = NULL;
 	if (agoIsValidData(data, VX_TYPE_PYRAMID) && (index < data->u.pyr.levels) && !data->isNotFullyConfigured) {
 		img = data->children[index];
-		agoRetainData((AgoGraph *)data->ref.scope, img, true);
+		agoRetainData((AgoGraph *)data->ref.scope, img, true);
```
Comment (kiritigowda) [inline amd_openvx/openvx/api/vx_api.cpp]: check this

File: amd_openvx/openvx/api/vx_api.cpp [hunk 6]
```diff
@@ -6733,10 +6734,11 @@ VX_API_ENTRY vx_status VX_API_CALL vxQueryPyramid(vx_pyramid pyr, vx_enum attrib
 VX_API_ENTRY vx_image VX_API_CALL vxGetPyramidLevel(vx_pyramid pyr, vx_uint32 index)
 {
 	AgoData * data = (AgoData *)pyr;
+
```
Comment (kiritigowda) [inline amd_openvx/openvx/api/vx_api.cpp]: check this

File: amd_openvx/openvx/include/VX/vx_compatibility.h [hunk 7]
```diff
@@ -127,9 +127,6 @@
 #define VX_ARRAY_ATTRIBUTE_CAPACITY         VX_ARRAY_CAPACITY
 #define VX_ARRAY_ATTRIBUTE_ITEMSIZE         VX_ARRAY_ITEMSIZE
```
Comment (kiritigowda) [inline amd_openvx/openvx/include/VX/vx_compatibility.h:129]: check this

File: amd_openvx/openvx/include/vx_ext_amd.h [hunk 8]
```diff
@@ -67,6 +67,11 @@ THE SOFTWARE.
  */
 #define VX_GRAPH_ATTRIBUTE_STATE            VX_GRAPH_STATE
```
Comment (kiritigowda) [inline amd_openvx/openvx/include/vx_ext_amd.h:69]: check this

File: amd_openvx/openvx/ago/ago_kernel_api.cpp (hunk 3 above)
Comment (LakshmiKumar23) [reply amd_openvx/openvx/ago/ago_kernel_api.cpp:2218]: @hansely can you look at this?

File: amd_openvx/openvx/ago/ago_kernel_api.cpp (hunk 4 above)
Comment (LakshmiKumar23) [reply amd_openvx/openvx/ago/ago_kernel_api.cpp:19770]: @hansely Can you look at this?

File: amd_openvx/openvx/ago/ago_kernel_api.cpp (hunk 3 above)
Comment (hansely) [reply amd_openvx/openvx/ago/ago_kernel_api.cpp:2218]: They are actually constant across the nodes.

Comment (hansely) [reply amd_openvx/openvx/ago/ago_kernel_api.cpp:2218]: Fixed the previous one by removing bracket.

File: amd_openvx/openvx/ago/ago_kernel_api.cpp (hunk 4 above)
Comment (hansely) [reply amd_openvx/openvx/ago/ago_kernel_api.cpp:19770]: This is correct.

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**